test/test-dirty-index.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+. "test/testlib.sh"
+
+begin_test "dirty index (git-lfs/git-lfs#1726)"
+(
+  set -e
+  reponame="dirty-index-prep"
+  setup_remote_repo "$reponame"
+  clone_repo "$reponame" repo_before_all
+
+  git lfs track "*.dat"
+  git add .gitattributes
+  git commit -m "initial commit"
+
+  printf "contents" > a.dat
+  ln -s a.dat a.symlink.dat
```

Comment:
The reason why the test does not fail (as would be expect) on Windows is that on MSYS2 this line will not actually create a symbolic link unless you do `export MSYS=winsymlinks:nativestrict` beforehand. Right now it just copies the file.